api/s2n.h

```diff
@@ -57,15 +58,19 @@ extern int s2n_config_free_dhparams(struct s2n_config *config);
 extern int s2n_config_free_cert_chain_and_key(struct s2n_config *config);
 
 typedef int (*s2n_clock_time_nanoseconds) (void *, uint64_t *);
+typedef int (*cache_retrieve) (struct s2n_connection *conn, void *, const void *key, uint64_t key_size, void *value, uint64_t *value_size);
```

Comment:
I think these typedef's should be renamed to `s2n_cache_{retrive,store,delete}_fn` or `s2n_cache_{retrive,store,delete}_callback` so that these functions don't collide with any other caller functions.

Reply:
updated